custom_enumerable.rb

```diff
@@ -0,0 +1,116 @@
+
+# frozen_string_literal: true
+
+# rubocop:disable Metrics/PerceivedComplexity,Metrics/CyclomaticComplexity, Style/CaseEquality
+
+
+module Enumerable
+  def my_each
+      i = 0
+      if block_given?
+      while i < self.length 
+          yield(self[i])
+          i += 1
+      end
+      end
+  end
+
+  def my_each_with_index
+    i = 0
+    for i in i...self.length
+      yield(self[i], i)
+  end
+
+  def my_select
+    array = []
+    self.my_each { |i| array << array[i] if yield(i) }
+    end
+  end  
+
+  def my_all?
+    final = true
+    self.my_each { |i| true unless yield(i) }
+    final
+  end
+
+  def my_any?
```

Comment:
- [ ] should be identical and returns the same thing as ruby's [`any?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-any-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_any.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- [ ] when no block or argument is given returns true if at least one of the collection is not false or nil
```ruby
true_array = [nil, false, true, []]
true_array.my_any? == true_array.any? #true
```
- [ ]  when a class is passed as an argument returns true if at least one of the collection is a member of such class
```ruby
words = %w[dog door rod blade]
words.my_any?(Integer) == words.any?(Integer) #true
```
- [ ] when a Regex is passed as an argument returns false if none of the collection matches the Regex
```ruby
words.my_any?(/z/) == words.any?(/z/) #true
```
- [ ] when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern
```ruby
words.my_any?('cat') == words.any?('cat') #true
```
- [ ] returns true if the block ever returns a value other than false or nil
```ruby
true_block = proc { |num| num <= 9 } 
false_block = proc { |num| num > 9 } 
array.my_any?(&true_block) == array.any?(&true_block) #true
array.my_any?(&false_block) == array.any?(&false_block) #true
```